### DISCUSSION TITLE
Column level security?

### DISCUSSION THREAD

**Discussion by riancz** [supabase/supabase#4744] [Questions] [answered]
I wonder, how do you guys deal with access to certain tables but **allowing only certain columns to be visible**.

**Example:**
You have table "profiles"
You want all users to get usernames and profile pics of other users, but you store more columns there like email which are suppose to be private.

I could make a view with only public data and then allow users to select all. The problem is views RLS is not supported in the GUI and I don't know if it has something to do with it not being advised to use.

Is there a better way to deal with it other than making another table?

**Comment by GaryAustin1** [comment, marked as answer]
Postgres has column grants.
I've used this article for reference before:
https://www.enterprisedb.com/postgres-tutorials/how-implement-column-and-row-level-security-postgresql
edit: a supabase oriented discussion on it: https://github.com/supabase/supabase/discussions/346#discussioncomment-182629

**Comment by riancz** [comment]
Thank you ☺️
